test/unit/storage/entities/account.js

```diff
@@ -1113,4 +1117,136 @@ describe('Account', () => {
 			});
 		});
 	});
+
+	describe('countDuplicatedDelegates()', () => {
+		it('should use the correct SQL no with parameter', async () => {
+			sinonSandbox.spy(adapter, 'executeFile');
+			await AccountEntity.countDuplicatedDelegates();
+
+			expect(adapter.executeFile).to.be.calledOnce;
+			expect(adapter.executeFile).to.be.calledWith(
+				SQLs.countDuplicatedDelegates,
+				{},
+				{ expectedResultCount: 1 }
+			);
+		});
+
+		it('should return zero if no delegate records available', async () => {
+			const block = seeder.getLastBlock();
+
+			const trs1 = new transactionsFixtures.Transaction({
+				blockId: block.id,
+				type: 2,
+			});
+			const trs2 = new transactionsFixtures.Transaction({
+				blockId: block.id,
+				type: 2,
+			});
+			await TransactionEntity.create([trs1, trs2]);
+
+			const delegates = await adapter.execute('SELECT * from delegates');
+
+			// As we created two delegate transactions
+			expect(delegates).to.have.lengthOf(2);
+
+			const result = await AccountEntity.countDuplicatedDelegates();
+
+			expect(result).to.be.eql(0);
+		});
+
+		it('should return zero if there are delegates but no duplicates', async () => {
+			const result = await AccountEntity.countDuplicatedDelegates();
+
+			expect(result).to.be.eql(0);
+		});
+
+		it('should return integer value of duplicate delegates', async () => {
+			const block = seeder.getLastBlock();
+
+			const trs1 = new transactionsFixtures.Transaction({
```

Comment:
It's redundancy here - `trs1` and `trs2` are getting inserted, but their effect is not checked - `countDuplicatedDelegates()` returns 2 only because of the direct insert of 2 duplicated delegate names in the next line:
```
	await adapter.execute(
					'INSERT INTO delegates ("transactionId", "username") VALUES ($1, $2)',
					[delegate.transactionId, username]
				);
```
Insert of `trs1` and `trs2` can be skipped or they both should be inserted with duplicated "username" passed in transaction.asset.username. 

Btw constraint on `delegates` table which should prevent of inserting duplicated username is preventing only from inserting the same delegate with the same transaction id - 

`ALTER TABLE delegates ADD CONSTRAINT delegates_unique UNIQUE ("username", "transactionId");`

Do you think a separate constraint should be added to prevent duplicated delegates cases? Then `countDuplicatedDelegates()` would not be needed.

Reply:
To add a duplicated delegate, we need a valid `transactionId`, so first we process tow valid transactions, and then based on their id we added a duplicate name entry. That's exactly what data sate we need to test the SQL written  for `countDuplicateDelegates`. And that was actually migrated from existing test case. 

Yes the constraint can be added to avoid duplicate. But I am not sure about historical data, I suspect there was some issue in data so why this check was added. 

In any case, would not suggest to make this behavioral change int he `modules/loader` at this moment.

Reply:
Why I didn't processed `trs1` and trs2` to be duplicate of each other, to check if our query can work perfectly to find multiple duplicate count.

Reply:
I have opened the issue for that https://github.com/LiskHQ/lisk/issues/2746.